frontend-react/src/ScrollToTop.tsx

```diff
@@ -0,0 +1,18 @@
+import { useEffect } from "react";
+import { useLocation } from "react-router-dom";
+
+export const ScrollToTop = () => {
+    const { pathname } = useLocation();
+
+    useEffect(() => {
+        const canControlScrollRestoration =
+            "scrollRestoration" in window.history;
+        if (canControlScrollRestoration) {
```

Comment:
Could you explain this logic to a n00b? I'm not familiar with `scrollRestoration` as a value of `window.history`, and not sure where we _shouldn't_ have it scroll to top

Reply:
It seems to be a recent thing that browsers are implementing. See https://developer.mozilla.org/en-US/docs/Web/API/History/scrollRestoration

Reply:
Does setting the scroll restoration mode need to be applied whenever the `pathname` changes or can it be applied on mount?

Reply:
I haven't used scroll restoration either so I may be lacking context over its usage, but do we need it if we're explicitly calling `window.scrollTo(0, 0);` on each navigation below?

Reply:
I moved the scrollRestoration portion to an useEffect init block